packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -146,44 +141,86 @@ const useTooltip = (options: TooltipOptions = {}) => {
       middleware.inline && inline(),
       middleware.hide && hide(),
     ].filter(Boolean),
+    ...(reference && { elements: { reference } }),
   });
 
-  const context = data.context;
+  const ariaProps = React.useMemo(
+    () =>
+      ariaStrategy === 'description'
+        ? { 'aria-describedby': id }
+        : ariaStrategy === 'label'
+        ? { 'aria-labelledby': id }
+        : {},
+    [ariaStrategy, id],
+  );
+
+  React.useEffect(() => {
+    if (!reference) {
+      return;
+    }
+
+    const oldValues: Record<string, string | null> = {}; // for cleanup
+    Object.entries(ariaProps).forEach(([key, value]) => {
+      oldValues[key] = reference.getAttribute(key);
+      reference.setAttribute(key, value);
```

Comment:
@gretanausedaite I was able to fix focus event handlers for `reference` prop using this code. What do you think?

```diff
+ /** e.g. onPointerDown --> pointerdown */
+ const domEventName = (e: string) => e.toLowerCase().substring(2);

  const oldValues: Record<string, string | null> = {};
  Object.entries({ ...ariaProps, ...interactions.getReferenceProps() }).forEach(([key, value]) => {
+   if (typeof value === 'function') {
+     reference.addEventListener(domEventName(key), value);
    } else if (value) {
      oldValues[key] = reference.getAttribute(key);
      reference.setAttribute(key, value);
    }
  });
```

Reply:
[`da3c1f5`](https://github.com/iTwin/iTwinUI/pull/1535/commits/da3c1f5809d8d0c1f77bf12a55070e4bbaeff6a3)